drivers/memc/Kconfig.stm32

```diff
@@ -1,9 +1,12 @@
 # Copyright (c) 2020 Teslabs Engineering S.L.
 # SPDX-License-Identifier: Apache-2.0
 
+DT_COMPAT_ST_STM32_FMC := st,stm32-fmc
+
 config MEMC_STM32
 	bool "Enable STM32 Flexible Memory Controller (FMC)"
-	depends on SOC_SERIES_STM32H7X
+	depends on SOC_SERIES_STM32H7X || SOC_SERIES_STM32F4X
```

Comment:
Seems we're back to square one.
These 2 lines should be redundant. Did you face a specific issue ?

Reply:
Well I still removed the block in `Kconfig.defconfig.series`, so these are not redundant anymore now and kind of make sense. The only issue I see now is maintaing this list (likely not a big deal for MEMC specifically).

Reply:
Well, the use of compatible macro here is to be able to use device tree as the main source of HW support.
So I prefer we keep the line in `soc/arm/st_stm32/common/Kconfig.defconfig.series` and remove this list.
Also this consistent with what we do for (most of) other drivers.

Reply:
Most drivers have, in fact, `depends on SOC_FAMILY_STM32`

Reply:
I liked the `depends on $(dt_...)` cause it's representative of the fact that the driver code itself depends on the node being defined and enabled, so that has the lowest chances of resulting in configuration that does not build. Are you aware of a specific Zephyr wide guideline to follow or are we just trying to be coherent?

Reply:
> Are you aware of a specific Zephyr wide guideline to follow or are we just trying to be coherent?

In general, we're trying to minimize use of Kconfig when possible and use device tree as main source of hw support when possible.

 > Most drivers have, in fact, depends on SOC_FAMILY_STM32, but that will show the option for series that do not have FMC available.

I agree this is not ideal. Maybe I'm minimizing the impact because I'm not a major menuconfig user, but in another hand I haven't seen a lot of complains on this particular topic.
Anyway, we should provide some better solution, but this should rely on device tree rather than Kconfig, as with Kconfig we should anyway maintain a list of SERIES/SOC symbols, which I don't think is reasonable.
In the end, using device tree, we should be able to both detect:
- what node is available
- what node is enabled

Reply:
Sounds like

```
depends on $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FMC))
default y
```

to me. That's what I'd like anyway. :-)

Reply:
> depends on $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FMC))
> default y

IIC that's equivalent to `default y $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FMC))`.
Also, can you revert the lines in `soc/arm/st_stm32/common/Kconfig.defconfig.series` ?

Then we'll be on par with existing practice in other drivers and we'll be able to merge.

About controlling more strictly drivers symbols, I'm not opposed to investigate and do some clean up.
But let's see if this can be addressed globally (at least on STM32 level), identifying exactly the issue, the impact, and how we can use dts to fix that.

Reply:
Yeah let's address it separately, so I reverted the `soc/arm/st_stm32/common/Kconfig.defconfig.series`, leaving these are they are now. LMK if that's what you mean.

Reply:
Let's alos 

> Yeah let's address it separately, so I reverted the `soc/arm/st_stm32/common/Kconfig.defconfig.series`, leaving these are they are now.

This and then we miss removing the following:
```
	depends on SOC_SERIES_STM32H7X || SOC_SERIES_STM32F4X
```

Reply:
Alright, done.